x/community/keeper/rewards.go

```diff
@@ -0,0 +1,45 @@
+package keeper
+
+import (
+	sdkmath "cosmossdk.io/math"
+)
+
+const SecondsPerYear = 365 * 24 * 3600
+
+// StakingRewardCalculator wraps data and calculation for determining staking rewards
+// It assumes that staking comes from one of two sources depending on if inflation is enabled or not.
+type StakingRewardCalculator struct {
```

Comment:
Generally, not a fan of use case style structs like this unless we have to satisfy an interface or the caller doesn't have access to some of the calculation inputs or dependencies. We are also creating a large interface here with public data members and several public methods that aren't used by non-test or test code.

I'd lean towards keeping it simple, pure, and less rigid with something like:
```go
func CalculateStakingAnnualPercentage(totalSupply, totalBonded sdkmath.Int, inflationRate, communityTax, rewardsPerSecond sdkmath.LegacyDec) sdkmath.LegacyDec {
	// no rewards are given if no tokens are bonded, in addition avoid division by zero
	if totalBonded.IsZero() {
		return sdkmath.LegacyZeroDec()
	}

	// the percetage of inflationRate * totalSupply tokens that are distrubed to stakers
	percentInflationDistributedToStakers := sdkmath.LegacyOneDec().Sub(communityTax)

	// the total amount of tokens distributed to stakers in a year
	amountGivenPerYear := inflationRate.
		MulInt(totalSupply).Mul(percentInflationDistributedToStakers).  // portion provided by inflation via mint & distribution modules
		Add(rewardsPerSecond.Mul(sdkmath.LegacyNewDec(SecondsPerYear))) // portion provided by community module

	// divide by bonded tokens or invested tokens to get the percent return
	return amountGivenPerYear.QuoInt(totalBonded)
}

```
Which also allows us to avoid duplicate zero checks and provide slightly higher precision by adding the two portions of rewards before dividing.